### DISCUSSION TITLE
Schemad or structured configuration for prompt

### DISCUSSION THREAD

**Discussion by jasonslyvia** [langfuse/langfuse#2699] [Ideas]
### Describe the feature or potential improvement

Currently we have a `config` text field for prompt written in JSON, it would be nice to have a more structured configuration user interface to setup configuration. 

![CleanShot 2024-07-22 at 10 13 33@2x](https://github.com/user-attachments/assets/bab4034c-5201-40e2-93d1-7447fff364aa)

My specific use case is to setup different model and provider for specific prompt, thus being able to choose them in the runtime.  It would be even nicer to have `LLM Provider` and `Model` to be configured or assosiated to prompt as first-class configuration.

**Comment by marcklingen** [comment]
Thanks for sharing!

Enforcing the schema would also be useful for dataset input/output as it can be edited via the UI.
